applications/CompressiblePotentialFlowApplication/python_scripts/potential_flow_solver.py

```diff
@@ -68,9 +105,15 @@ def AddVariables(self):
         # Degrees of freedom
         self.main_model_part.AddNodalSolutionStepVariable(KCPFApp.VELOCITY_POTENTIAL)
         self.main_model_part.AddNodalSolutionStepVariable(KCPFApp.AUXILIARY_VELOCITY_POTENTIAL)
-
+        # Embedded variables
+        self.main_model_part.AddNodalSolutionStepVariable(KCPFApp.GEOMETRY_DISTANCE)
         # Kratos variables
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.FLAG_VARIABLE) # Required for variational_distance_process
         self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.DISTANCE)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.DISTANCE_GRADIENT)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.NODAL_H) # Required for modify_distance_process
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.VELOCITY) # Required for modify_distance_process
```

Comment:
You need these variables because the modify distance process fixes them to zero in the fully negative and continuous distance case. Am I wrong?

In case you don't need them (I think you need at least the velocity to do the post-process), we should modify the modify distance process to skip this operation in case is not needed since this implies a useless memory overhead. Would you mind open an issue describing the situation?

Reply:
Yes, you are right! I will open an issue.

For now we are just using the VELOCITY at the Gauss Points, so actually we dont need to define it as a historical nodal variable.